doc/authentication.md

```diff
@@ -0,0 +1,73 @@
+Requiring authentication
+========================
+
+In some cases, it might be needed for security or legal reasons to secure the
+projects' code available through CodeCompass against unauthorised access.
+
+To enable this feature, an `authentication.json` file should be created under
```

Comment:
And there is still the issue of the "session_manager.py" (or whichever file) parsing and handling stuff that are not related to sessions...

https://github.com/Ericsson/codechecker/blob/65aa0c90a4f5d8a1d857e4ef1570045419c65266/web/server/codechecker_server/session_manager.py#L173-L179

Reply:
I believe in the future we can live with separation of concerns and if there is a new feature which configures different aspects of the server, it could get a different parser, handler, and file for it.

Which also allows separation of concerns between the individual units of C++ code in the webserver that handle parsing and configuring stuff. Unlike Python, we can't really add new values without adding them to a header, having to recompile the code affected, changing objects' size - which might also affect the size of objects passed between functions.